packages/amplify-category-api/src/provider-utils/awscloudformation/utils/rds-secrets/multi-env-database-secrets.ts

```diff
@@ -0,0 +1,27 @@
+import { $TSContext } from 'amplify-cli-core';
+import { storeConnectionSecrets, getExistingConnectionSecrets } from '../../utils/rds-secrets/database-secrets';
+import { printer } from 'amplify-prompts';
+
+type EnvironmentInfo = {
+  isNewEnv: boolean,
+  sourceEnv: string,
+  yesFlagSet: boolean,
+  envName: string
+};
+
+export const configureMultiEnvDBSecrets = async (context: $TSContext, database: string, apiName: string, envInfo: EnvironmentInfo) => {
+  // For existing environments, the secrets are already set in parameter store
+  if(!envInfo.isNewEnv) {
+    return;
+  }
+
+  // For new environments, the secrets are copied over from the source environment.
+  const secrets = await getExistingConnectionSecrets(context, database, apiName, envInfo.sourceEnv);
+  if (!secrets) {
```

Comment:
Can we add an E2E test for the secret workflow at this state if possible?

Reply:
Can I do it in a different PR along with the `generate-schema` command tests as well?